### ISSUE TITLE
Add http headers

### ISSUE THREAD

**Issue by purplescorpion1** [imrsaleh/Easy-M3U-Playlist-Creator#4]
There needs to be a box to add in http headers such as
#EXTVLCOPT:http-referrer=
#EXTVLCOPT:http-user-agent=

**Comment by imrsaleh** [comment]
try adding it at the end of .m3u8/.mpd using |Referrer="http://example.com/"

for example:
`https://live-hls-web-aja-fa.getaj.net/AJA/index.m3u8|Referer="http://example.com/"&user-agent="Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/129.0.0.0 Safari/537.36"`

**Comment by purplescorpion1** [comment]
m3u files don't work like that 

You need to pass headers like this (for vlc or clients that support extvlcopt) 
Other players you would need to inject using something like mitmproxy or other methods. Not via the stream link itself
```
#EXTM3U
#EXTINF:-1 tvg-name="channel" tvg-id="epgforchannel" tvg-logo="https://domain.com/logo.png" group-title="channelgroup",channel name
#EXTVLCOPT:http-referrer=https://domain.com
#EXTVLCOPT:http-user-agent=Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:128.0) Gecko/20100101 Firefox/128.0
https://domain.com/streamlink.m3u8
```

**Comment by imrsaleh** [comment]
> m3u files don't work like that
> 
> You need to pass headers like this (for vlc or clients that support extvlcopt) Other players you would need to inject using something like mitmproxy or other methods. Not via the stream link itself
> 
> ```
> #EXTM3U
> #EXTINF:-1 tvg-name="channel" tvg-id="epgforchannel" tvg-logo="https://domain.com/logo.png" group-title="channelgroup",channel name
> #EXTVLCOPT:http-referrer=https://domain.com
> #EXTVLCOPT:http-user-agent=Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:128.0) Gecko/20100101 Firefox/128.0
> https://domain.com/streamlink.m3u8
> ```

Yes you are right "vlc" doesn't support that, but it works using "tivimate/ott navigator"
Stay tuned for the latest updates and thanks for your suggestion
